R/merge_sce_list.R

```diff
@@ -263,3 +335,99 @@ prepare_sce_for_merge <- function(sce,
   # return the processed SCE
   return(sce)
 }
+
+
+
+#' Prepare altExps for merge and create a list of merged altExps for each altExp name
+#'
+#'
+#' @param altexp_list List of altexps to merge
+#' @param merged_colnames Vector of column names (`{sce_name}-{barcode}`) to include
+#'   in the final merged altExp. This vector includes _all_ SCEs, not only those
+#'   with this altExp name.
+#'
+#' @return A list of merged altExps to include the final merged SCE object
+create_merged_altexp <- function(
+    altexp_list,
+    merged_colnames) {
+
+  # Create vector of all features
+  # this order will be used for the final assay matrix/ces
+  altexp_features <- altexp_list |>
+    purrr::map(rownames) |>
+    purrr::reduce(union)
+
+  # Determine which assays are present for this altexp_name. We'll need a matrix
+  # for each of these in the final merged object
+  altexp_assay_names <- altexp_list |>
+    purrr::map(assayNames) |>
+    purrr::reduce(union)
+
+  # Create new merged assay matrices
+  new_assays <- altexp_assay_names |>
+    purrr::map(
+      build_new_altexp_assay,
+      altexp_list,
+      altexp_features,
+      merged_colnames
+    )
+  names(new_assays) <- altexp_assay_names
+
+
+  # Create merged altExp
+  merged_altexp <- SingleCellExperiment(assays = new_assays)
+
+  # TODO: Add rowData (and colData, if we need it) to merged_altexp
+
+  return(merged_altexp)
+
+}
+
+
+
+
+#' Build a new sparse matrix for merging altExps
+#'
+#' @param assay_name Name of assay of interest (e.g., "counts")
+#' @param altexp_list List of altExps which should be included in the new matrix
+#' @param merged_row_names Vector of matrix row names, corresponding to the full
+#'   set of features for this altExp
+#' @param merged_col_names Vector of matrix column names, corresponding to all cells
+#'   which will be in the final merged altExp
+#'
+#' @return Sparse matrix
+build_new_altexp_assay <- function(
+    assay_name,
+    altexp_list,
+    merged_row_names,
+    merged_col_names) {
```

Comment:
```suggestion
    merged_rownames,
    merged_colnames) {
```
This is minor, but everywhere else you use `colnames`